### PR TITLE
Alphabetically sort organized declarations by name pattern

### PR DESCRIPTION
**Pull request by oiuhr**
Added an ability to sort declarations alphabetically within `organizeDeclaration` by their name pattern instead of `// swiftformat:sort` marks.

Motivation: in our project we have 1000+ enums named `Constants` with static properties which we wanted to see sorted alphabetically — and marking all of them with sorting comments is a bit tedious, if not straight up rude 🥹

New option `sortedpatterns` can (and should?) be shared with existing `sortDeclaration` rule.

### CODE REVIEW

File: Tests/RulesTests+Organization.swift [hunk 1]
```diff
@@ -4002,6 +4002,127 @@ class OrganizationTests: RulesTests {
...
+        }
+        """
+
+        let output = """
+        enum FeatureFlags {
+            case barFeature
+            case fooFeature
+            case upsellA
+
+            case upsellB
```
Comment (oiuhr) [inline Tests/RulesTests+Organization.swift:4031]: Were not using sorting in our project (yet), so I`m not really familiar with sorting within `organizeDeclarations` rule — but in current state behavior between this case and `sortDeclarations` is clearly differ. For example, `sortDeclarations` won`t leave newline before `upsellB` case. 

Shall we share an implementation of sorting between them or there is some pitfalls in that regard?

File: Sources/OptionDescriptor.swift [hunk 2]
```diff
@@ -859,6 +859,12 @@ struct _Descriptors {
         help: "Organize declarations by \"visibility\" (default) or \"type\"",
         keyPath: \.organizationMode
     )
+    let alphabeticallySortedDeclarationPatterns = OptionDescriptor(
+        argumentName: "sortedpatterns",
+        displayName: "Declaration Name Patterns To Sort Alphabetically",
```
Comment (calda) [inline Sources/OptionDescriptor.swift:864]: Maybe we should include an example "pattern" to help demonstrate the syntax? I assume you can use `*`s?

File: Sources/Rules.swift [hunk 3]
```diff
@@ -5656,8 +5656,12 @@ public struct _FormatRules {
         runOnceOnly: true,
         disabledByDefault: true,
         orderAfter: ["extensionAccessControl", "redundantFileprivate"],
-        options: ["categorymark", "markcategories", "beforemarks", "lifecycle", "organizetypes",
-                  "structthreshold", "classthreshold", "enumthreshold", "extensionlength", "organizationmode"],
+        options: [
+            "categorymark", "markcategories", "beforemarks",
+            "lifecycle", "organizetypes", "structthreshold", "classthreshold",
+            "enumthreshold", "extensionlength", "organizationmode",
+            "sortedpatterns",
```
Comment (calda) [inline Sources/Rules.swift:5663]: Could you also add support for this option to the `sortDeclarations` rule, so this functionality is supported even when the `organizeDeclarations` rule isn't enabled?

File: Tests/RulesTests+Organization.swift (hunk 1 above)
Comment (calda) [reply Tests/RulesTests+Organization.swift:4031]: I think this is fine -- this is actually the trailing blank line after the `case upsellA`, which is followed by a blank line in the input code. It's difficult to know how to handle blank lines so a best-effort like this is fine imo

**Review by calda — COMMENTED** [review]
I think this is a great idea, thanks! 

Sorting is primarily implemented in the `sortDeclarations` rule, so we should support this option there as well.

**Comment by nicklockwood** [comment]
@oiuhr oops, sorry - I tried to fix the conflicts in your PR after rebasing develop and somehow borked the PR so it got closed. I've landed your feature on the develop branch so it doesn't get lost - if you were planning on making any further changes please open a new PR.

**Pull request closed without merging**